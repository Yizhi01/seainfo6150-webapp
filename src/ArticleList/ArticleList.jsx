import React from "react";
import PropTypes from "prop-types";
import ArticleListItem from "../ArticleListItem/ArticleListItem";
import styles from "./ArticleList.module.css";

const ArticleList = (props) => {
    let displayContent;
    if (props.articlelist.length) {
        displayContent = (
            <ul className={styles.list}>
                {props.articlelist.map((article) => (<ArticleListItem article={article} key={article.slug}/>))}
            </ul>
        )
    }else {
        displayContent = <div>You have no data!</div>;
    }
    
    return (
        <div className={styles.container}>
            {displayContent}
        </div>
    );
};

ArticleList.propTypes = {
    articlelist: PropTypes.array.isRequired
}

export default ArticleList;