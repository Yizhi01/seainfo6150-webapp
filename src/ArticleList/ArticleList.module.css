body {
    font-family: Arial, Helvetica, sans-serif;
}

.container {
    background-color: lightgray;
    box-shadow: 0 0 20px red;
    border: 1px dashed grey;
    border-radius: 10px;
    margin: 20px 20px;
}

.list {
    margin: auto 0;
    padding: 0 20px;
}