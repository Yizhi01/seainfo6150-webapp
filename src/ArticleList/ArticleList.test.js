import React from "react";
import ArticleList from "./ArticleList";
import { fireEvent } from "@testing-library/react";

describe("ArrticleList tests", () => {
  it("renders correctly", () => {
    const testList = [{"title" : "Story1", "slug" : "Test1"}, {"title" : "Story2u", "slug" : "Test2"}];
    const { container } = render(
      <ArticleList articles={testList} />
    );
    expect(container).toMatchSnapshot();
  });
});