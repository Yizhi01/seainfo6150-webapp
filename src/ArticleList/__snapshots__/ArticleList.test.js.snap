// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrticleList tests renders correctly 1`] = `
<div>
  <div>
    <ul
      class="container"
    >
      <li
        class="container"
      >
        <article
          class="article"
        >
          <div
            class="wrapper"
          >
            <h2
              class="title"
            >
              Story1
            </h2>
          </div>
          <button
            class="button"
          >
            Show more
          </button>
        </article>
      </li>
      <li
        class="container"
      >
        <article
          class="article"
        >
          <div
            class="wrapper"
          >
            <h2
              class="title"
            >
              Story2u
            </h2>
          </div>
          <button
            class="button"
          >
            Show more
          </button>
        </article>
      </li>
    </ul>
  </div>
</div>
`;
